lib/gruf/integrations/rails/railtie.rb

```diff
@@ -23,10 +23,10 @@ module Rails
       #
       class Railtie < ::Rails::Railtie
         initializer 'gruf.initializer' do |app|
-          config.before_configuration do
+          config.after_initialize do
```

Comment:
Rails is pretty finicky with initialization order changes between versions. @salmanasiddiqui - I think we're probably going to need to see some proof of life here that this change doesn't break Rails 7.0, 7.1, 7.2, and 8.0 installations.

Reply:
I think eager load paths are frozen at the point of `after_initialize` so you won't be able to mutate it

Reply:
thanks for reviewing this, Im trying to figure if there is a good way to test this railtie. Any suggestions or relevant resources will be greatly appreciated

@zzak - `after_initialize` is used by gruf-rspec aswell, and I have tested this with Rails 7.1, it updated the autoloader correctly.
https://github.com/bigcommerce/gruf-rspec/blob/main/lib/gruf/rspec/railtie.rb#L25

Reply:
Have you tried EAGER_LOAD=true?

Reply:
yes, I did